### PR TITLE
Table styling update: Headers should appear to overhang above the scrollbar

### PR DESCRIPTION
**Pull request by msmithNI**
# Pull Request

## 🤨 Rationale

Followup PR for #865 - after feedback from Nate/Rick, we want the header row to appear to overhang above the table scroll bar.

See https://github.com/ni/nimble/issues/865#issuecomment-1410372829 for context and screenshots.

## 👩‍💻 Implementation

Instead of setting `margin-right` on the header container based on the scrollbar width (which left a transparent square in the top-right corner of the table bounds above the scrollbar), add an empty `div` to the end of the table container with that same width. Net result is that upper-right square appears with the same background color as the headers, and gets the overhang effect.

(Note: The reason why we need the margin in the first place is because otherwise, the column headers would no longer align vertically with the cells, once the scrollbar was showing.)

Before:  
TableScrollOverhangBefore

After:  
TableScrollOverhangAfter

## 🧪 Testing

Tested via Storybook.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by msmithNI** [comment]
@atmgrifter00 added you for awareness and since you own the Column Width HLD. Not sure if we'll need a different approach for this once we have resizable columns, I think this is the simplest approach for now though.

**Comment by atmgrifter00** [comment]
> @atmgrifter00 added you for awareness and since you own the Column Width HLD. Not sure if we'll need a different approach for this once we have resizable columns, I think this is the simplest approach for now though.

Well, the concern is that we ultimately (imminently even) expect to be moving to a `grid` layout for the elements in both a table row, and the header, using the `grid-template-columns` layout mechanism to set the size for each column. The `div` you are adding I suppose could be factored in for the header, but it's unclear to me how annoying that will be.

File: packages/nimble-components/src/table/template.ts [hunk 1]
```diff
@@ -28,13 +28,14 @@ const isTableColumn = (): ElementsFilter => {
...
         <div class="table-container">
-            <div role="rowgroup" class="header-container" style="margin-right: ${x => x.virtualizer.headerContainerMarginRight}px;">
+            <div role="rowgroup" class="header-container">
                 <div class="header-row" role="row">
                     ${repeat(x => x.columns, html<TableColumn>`
                         <${DesignSystem.tagFor(TableHeader)} class="header">
                             ${x => x.textContent}
                         </${DesignSystem.tagFor(TableHeader)}>
                     `)}
+                    <div style="width: ${x => x.virtualizer.headerContainerMarginRight}px;"></div>
```
Comment (rajsite) [inline packages/nimble-components/src/table/template.ts:38]: Maybe add a template comment describing why we need it (i.e. one of those `${'' /* Spacer div to extend the header space over the scrollbar */}`) or a descriptive class like `header-scrollbar-spacer` or something.

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/table/template.ts (hunk 1 above)
Comment (msmithNI) [reply packages/nimble-components/src/table/template.ts]: Added the CSS class.

**Pull request merged**